packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Statements/XmlDocStatement.cs

```diff
@@ -25,7 +25,7 @@ public XmlDocStatement(string tagName, IEnumerable<FormattableString> lines, par
         {
         }
 
-        public XmlDocStatement(string startTag, string endTag, IEnumerable<FormattableString> lines, params XmlDocStatement[] innerStatements)
+        public XmlDocStatement(FormattableString startTag, FormattableString endTag, IEnumerable<FormattableString> lines, params XmlDocStatement[] innerStatements)
```

Comment:
the reason I did not create another overload for this method but have to change the signature here is that the csharp compiler will not treat `$"something"` as a `FormattableString` when we have both overloads.
For example, if we have
```
public void DoSomething(string str) {}
public void DoSomething(FormattableString str) {}
```
and we call it by
```
DoSomething($"{a}something");
```
this will actually call the `string` version, unless we explicitly cast the interpolation string into an instance of `FormattableString`.
Therefore I think if we keep the overload that takes string parameters, it would be very confusing when someone wants to call the FormattableString version, and it might cause some weird issues which are hard to debug.